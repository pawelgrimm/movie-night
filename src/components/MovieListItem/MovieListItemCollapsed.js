import React from "react";
import ReactImageFallback from "react-image-fallback";
import { formatReleaseYear } from "../../utils/theMovieDb";

const MovieListItemCollapsed = ({
  title,
  releaseDate,
  genres,
  runtime,
  voteAverage,
  posterPath,
  onClick,
  buttons = [],
}) => {
  return (
    <div className="movie-item list-item--hoverable" onClick={onClick}>
      <ReactImageFallback
        src={`https://image.tmdb.org/t/p/w92${posterPath}`}
        fallbackImage="/images/default_poster.png"
        initialImage="/images/loader.gif"
        alt={`poster image for ${title}`}
        className="movie-item__poster movie-item__poster--small"
      />
      <div className="movie-item__info--small">
        <h3 className="movie-item__title">{title}</h3>
        <div className="movie-item__meta">
          <span>{formatReleaseYear(releaseDate)}</span>
          <span>{genres?.map((genre) => genre.name).join(", ")}</span>
          <span>{runtime} m</span>
          <span>{voteAverage}/10</span>
        </div>
      </div>
      {React.Children.toArray(buttons)}
    </div>
  );
};

export default MovieListItemCollapsed;
