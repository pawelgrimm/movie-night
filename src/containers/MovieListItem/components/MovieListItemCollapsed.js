import React from "react";
import ReactImageFallback from "react-image-fallback";
import { formatReleaseYear } from "../../../utils/theMovieDb";
import Loader from "../../../components/Loader/Loader";

const MovieListItemCollapsed = ({
  id,
  movie,
  onClick,
  renderAdditionalItems,
}) => {
  if (!movie) {
    return <Loader />;
  }
  const {
    poster_path,
    title,
    release_date,
    genres,
    runtime,
    vote_average,
  } = movie;

  return (
    <div className="movie-item list-item--hoverable" onClick={onClick}>
      <ReactImageFallback
        src={`https://image.tmdb.org/t/p/w92${poster_path}`}
        fallbackImage="/images/default_poster.png"
        initialImage="/images/loader.gif"
        alt={`poster image for ${title}`}
        className="movie-item__poster movie-item__poster--small"
      />
      <div className="movie-item__info--small">
        <h3 className="movie-item__title">{title}</h3>
        <div className="movie-item__meta">
          <span>{formatReleaseYear(release_date)}</span>
          <span>{genres?.map((genre) => genre.name).join(", ")}</span>
          <span className="movie-item__meta__runtime">
            {runtime ? runtime + " m" : ""}
          </span>
          <span>{vote_average ? vote_average + "/10" : ""}</span>
        </div>
      </div>
      {renderAdditionalItems(id)}
    </div>
  );
};

export default MovieListItemCollapsed;
