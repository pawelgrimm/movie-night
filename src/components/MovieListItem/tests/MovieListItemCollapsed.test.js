import { expect } from "@jest/globals";
import { shallow } from "enzyme";
import React from "react";
import { movies } from "../../../global/tests/fixtures/fixtures";
import MovieListItemCollapsed from "../MovieListItemCollapsed";
import { Trash2 as TrashIcon } from "react-feather";

test("should render MovieListItemCollapsed correctly with no buttons", () => {
  const movie = movies[2];
  const wrapper = shallow(
    <MovieListItemCollapsed
      title={movie.title}
      rating={movie.rating}
      releaseDate={movie.release_date}
      genres={movie.genres}
      runtime={movie.runtime}
      voteAverage={movie.vote_average}
      posterPath={movie.poster_path}
      onClick={() => {}}
      buttons={undefined}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render MovieListItemCollapsed correctly with buttons", () => {
  const movie = movies[2];
  const buttons = [<button />, <button />];

  const wrapper = shallow(
    <MovieListItemCollapsed
      title={movie.title}
      rating={movie.rating}
      releaseDate={movie.release_date}
      genres={movie.genres}
      runtime={movie.runtime}
      voteAverage={movie.vote_average}
      posterPath={movie.poster_path}
      onClick={() => {}}
      buttons={buttons}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("button")).toHaveLength(buttons.length);
});
