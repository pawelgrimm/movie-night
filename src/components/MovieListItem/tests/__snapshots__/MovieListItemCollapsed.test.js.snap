// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieListItemCollapsed correctly with buttons 1`] = `
<div
  className="list-item list-item--hover-effect movie-item"
  onClick={[Function]}
>
  <ReactImageFallback
    alt="poster image for Before Sunset"
    className="movie-item__poster movie-item__poster--small"
    fallbackImage="/images/default_poster.png"
    initialImage="/images/loader.gif"
    src="https://image.tmdb.org/t/p/w92/gycdE1ARByGQcK4fYR2mgpU6OO.jpg"
  />
  <div
    className="movie-item__info--small"
  >
    <h3
      className="movie-item__title"
    >
      Before Sunset
    </h3>
    <div
      className="movie-item__meta"
    >
      <span>
        2004
      </span>
      <span />
      <span>
         m
      </span>
      <span>
        7.8
        /10
      </span>
    </div>
  </div>
  <button
    key=".0"
  />
  <button
    key=".1"
  />
</div>
`;

exports[`should render MovieListItemCollapsed correctly with no buttons 1`] = `
<div
  className="list-item list-item--hover-effect movie-item"
  onClick={[Function]}
>
  <ReactImageFallback
    alt="poster image for Before Sunset"
    className="movie-item__poster movie-item__poster--small"
    fallbackImage="/images/default_poster.png"
    initialImage="/images/loader.gif"
    src="https://image.tmdb.org/t/p/w92/gycdE1ARByGQcK4fYR2mgpU6OO.jpg"
  />
  <div
    className="movie-item__info--small"
  >
    <h3
      className="movie-item__title"
    >
      Before Sunset
    </h3>
    <div
      className="movie-item__meta"
    >
      <span>
        2004
      </span>
      <span />
      <span>
         m
      </span>
      <span>
        7.8
        /10
      </span>
    </div>
  </div>
</div>
`;
